### DISCUSSION TITLE
No longer getting zone changes

### DISCUSSION THREAD

**Discussion by olddawgpowers** [Dilbert66/esphome-dsckeybus#128] [Q&A]
None of my zones are showing status changes.  It has been working fine until the last Eshome update.  I loaded a previous backup but no changes.  Some of my log with some new errors
[15:06:29][D][text_sensor:064]: 'line1 Partition 1 (ln1_1)': Sending state 'System is Ready'
[15:06:29][D][text_sensor:064]: 'line2 Partition 1 (ln2_1)': Sending state 'Ready to Arm <>'
[15:06:29][D][text_sensor:064]: 'System Status (ss)': Sending state 'online'
[15:06:29][D][text_sensor:064]: 'Partition 1 Status (ps_1)': Sending state 'disarmed'
[15:06:29][D][text_sensor:064]: 'zone status (zs)': Sending state ''
[15:06:29][D][text_sensor:064]: 'Trouble Msg (tr_msg)': Sending state 'AC '
[15:06:29][W][component:232]: Component dsc_alarm_panel took a long time for an operation (71 ms).
[15:06:29][W][component:233]: Components should block for at most 30 ms.
[15:06:36][I][Paneldata:751]: 5D: 5D 00 00 00 00 00 00 5D 00 00 00 00 00 00 00 00  (58)
[15:06:40][I][Paneldata:751]: 11: 11 00 AA AA AA AA AA 2A 00 00 00 00 00 00 00 00  (55)
[15:06:40][I][Moduledata:751]: 11: FF 01 3F FC FF F3 CF 3F 00 00 00 00 00 00 00 00  (55)
[15:06:51][I][Paneldata:751]: 27: 27 00 81 01 91 C7 08 09 00 00 00 00 00 00 00 00  (58)
[15:06:51][D][text_sensor:064]: 'zone status (zs)': Sending state 'OP:4'

**Comment by Dilbert66** [comment]
Should be ok now. Recompile.

**Comment by olddawgpowers** [comment]
Do you mean to recompile using ESPHome and Install?  I just did that and still get the same log warnings.  None of my sensors are changing states.  For instance when I open the Front Door the state does not change.

In my yaml I have this to get updates:

external_components:
  - source: ***@***.*** #uncomment to use github repository
  #- source:  my_components #uncomment to use local directory
    components: [dsc_alarm_panel,binary_sensor,text_sensor]
    refresh: 10min

I don't think any of the sensors change state

[cid:9e91c99a-293e-49e1-88cb-55c4f36b21ff]
________________________________

**Comment by Dilbert66** [comment]
Yes, I just meant to recompile and install.  Do you mean the "long time for operation" warnings? Those are normal. ESPHOME added those a while back. Those are fine.

Can you show me one of your zone entries from the yaml?

**Comment by olddawgpowers** [comment]
binary_sensor:
### zone setup ###
# zone status open/close for each zone
# zone id code = z+zone number
  - platform: template
    id: z1
    name: "Front Door(z1)"
    device_class: door
    publish_initial_state: true

  - platform: template
    id: z2
    name: "Garage Door(z2)"
    device_class: door
    publish_initial_state: true

  - platform: template
    id: z3
    name: "Back Door(z3)"
    device_class: door
    publish_initial_state: true

  - platform: template
    id: z4
    name: "Family Motion(z4)"
    device_class: motion
    publish_initial_state: true

  - platform: template
    id: z9
    name: "Smoke Detector(z9)"
    device_class: smoke
    publish_initial_state: true

  - platform: template
    id: z10
    name: "Front Bedroom Window(z10)"
    device_class: window
    publish_initial_state: true

  - platform: template
    id: z11
    name: "Middle Bedroom Window(z11)"
    device_class: window
    publish_initial_state: true

  - platform: template
    id: z12
    name: "Master Left Bedroom Window(z12)"
    device_class: window
    publish_initial_state: true

  - platform: template
    id: z13
    name: "Master Right Bedroom Window(z13)"
    publish_initial_state: true
    device_class: window

  - platform: template
    id: z14
    name: "Dining Motion(z14)"
    device_class: motion
    publish_initial_state: true


________________________________

**Comment by Dilbert66** [comment]
ok, I see the issue.  You don't have a space before the () brackets for the id code and my code was assuming a space before.  I've modifed the code to accept it with or without the space.   Recompile again and see how it goes.

**Comment by olddawgpowers** [comment]
That took care of it.  It worked fine until I updated today?   Would you prefer I update with a space?  I appreciate your help and this project.

________________________________

**Comment by Dilbert66** [comment]
You're on the dev version which has gone through a few changes.  You only noticed today because you recompiled and it pulled the latest version from github when it did that as it should.  You can leave your yaml as is. No need to change.  An alternative way to identify the sensor records is by adding the "id_code:" line as shown below..  Then you don't need the (id) on the name.  This is for those that don't want the id showing up in home assistant.
```
# zone status open/close for each zone 
# zone id code = z+zone number  
  - platform: template
    id_code: z1
    name: "Front door"
    device_class: door
    publish_initial_state: true   
```

**Comment by olddawgpowers** [comment]
Thanks again for all your hard work.

John
________________________________

**Comment by Dilbert66** [comment]
No worries! Glad it works for you.
